src/components/carousel/NumbersDotsPagination.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC, useCallback } from 'react';
+import { EmblaCarouselType } from 'embla-carousel-react';
+
+import { useMultiStyleConfig } from '@chakra-ui/react';
+
+import Flex from '../flex';
+import Box from '../box';
+import NumbersPaginationButton from './NumbersPaginationButton';
+import DotsPaginationButton from './DotsPaginationButton';
+
+type PaginationType = 'dot' | 'number';
+
+interface Props {
+  currentSlideIndex: number;
+  numberOfSlides: number;
+  mainCarousel?: EmblaCarouselType;
+  paginationType: PaginationType;
+}
+
+const NumbersDotsPagination: FC<Props> = ({
+  currentSlideIndex,
+  numberOfSlides,
+  mainCarousel,
+  paginationType,
+}) => {
+  const onClick = useCallback(
+    (index: number) => {
+      if (mainCarousel) {
+        mainCarousel.scrollTo(index);
+      }
+    },
+    [mainCarousel]
+  );
+  const pagination = Array.from({ length: numberOfSlides }, (_, i) => i);
+  const { dotsPaginationContainer } = useMultiStyleConfig('Carousel');
+
+  const renderPagination = (type: PaginationType) => {
+    switch (type) {
+      case 'dot':
+        return (
+          <Box aria-label="Dots Pagination" __css={dotsPaginationContainer}>
+            <Flex
```

Comment:
those two branches are identical. If you do the switch inside NumbersPaginationButton and pass along the variant, you don't need it

Reply:
not sure I follow